bff/pkg/prometheus/prometheus.go

```diff
@@ -250,36 +264,116 @@ func processVector(vector model.Vector) ([]map[string]interface{}, error) {
 	return jsonVector, nil
 }
 
-func processMatrix(matrix model.Matrix, isTimeSeriesQuery bool) ([]map[string]interface{}, error) {
+func processMatrix(matrix model.Matrix, isTimeSeriesQuery bool, legend string, limitLegend int) ([]map[string]interface{}, error) {
+
 	if len(matrix) == 0 {
 		return []map[string]interface{}{}, nil
 	}
 
-	var jsonMatrix []map[string]interface{}
-	for _, stream := range matrix {
-		metricMap := make(map[string]interface{})
-		for k, v := range stream.Metric {
-			metricMap[string(k)] = string(v)
-		}
+	if legend == "" || limitLegend == -1 || limitLegend > len(matrix) {
+		var jsonMatrix []map[string]interface{}
+		for _, stream := range matrix {
+			metricMap := make(map[string]interface{})
+			for k, v := range stream.Metric {
+				metricMap[string(k)] = string(v)
+			}
 
-		values := make([]map[string]interface{}, len(stream.Values))
-		for i, sample := range stream.Values {
-			values[i] = map[string]interface{}{
-				"timestamp": int64(sample.Timestamp),
-				"value":     float64(sample.Value),
+			values := make([]map[string]interface{}, len(stream.Values))
+			for i, sample := range stream.Values {
+				values[i] = map[string]interface{}{
+					"timestamp": int64(sample.Timestamp),
+					"value":     float64(sample.Value),
+				}
+
+				if !isTimeSeriesQuery {
+					break
+				}
 			}
 
-			if !isTimeSeriesQuery {
-				break
+			jsonStream := map[string]interface{}{
+				"metric": metricMap,
+				"values": values,
 			}
+			jsonMatrix = append(jsonMatrix, jsonStream)
+		}
+
+		return jsonMatrix, nil
+	}
+
+	type metricSum struct {
+		index int
+		sum   float64
+	}
+
+	var metricsSums []metricSum
+
+	for i, stream := range matrix {
+		sum := 0.0
+		for _, sample := range stream.Values {
```

Comment:
FYI: I'm not able to evaluate whether this code is correct or not.

Reply:
ill add a unit test